Plugin.pm

```diff
@@ -288,6 +288,34 @@ sub trackInfoMenu {
 
 	my $search = cstring($client, 'SEARCH');
 	my $items = [];
+
+	my $artists = $track->remote ? $remoteMeta->{artists} : undef;
+	my $album_id = $track->remote ? $remoteMeta->{album_details}->{id} : undef;
```

Comment:
You could get the album Id from a database track object by getting `$track->album->extid` or similar.

Reply:
`$track->album->extid` isn't there because `$track->album` is already undefined.   I can't do "similar" as I'm not familiar with the code and framework, nor Perl too much tbh.

Reply:
You'd only get `$track->album` for track stored in the local music library. I'd have to double check, but I was pretty sure it would then be `$track->album->extid` (obviously only if `$track->album` was defined)

Reply:
I have a different problem here.  The intention to use `remoteMeta` is that how I use `$artists` and obviously `albumId` and `trackId` rely on the fact that the ids are coming from TIDAL.  For track_id I actually grab the url and extract the id from there.  And since the url starts with tidal: (iirc)   this is safe.  Is this here also safe for album_id and the artists?  I don't think so as the "on TIDAL" should appear for every song (in the queue).  That question also holds for if I ever get the `extid`.  Is that an extid actually from TIDAL, or could it be from Deezer as well?

Reply:
The IDs can of course be anything. But they'd always start with `tidal:` for TIDAL, `deezer:` for Deezer etc. That's how LMS distinguishes them. See https://github.com/michaelherger/lms-plugin-tidal/blob/604a850987c15cc24dc49b83ba3c3207cb2e83ff/ProtocolHandler.pm#L30.

That said: `_getId()` doesn't cover items other than tracks, yet.

Reply:
Okay, I added a check if the ids I use are valid and come from Tidal.  What you cite here aren't ids but URI's though.  I use them here e.g. `favorites_url => 'tidal://album:' . $albumId,`.  

If there were `extid`s they would probably start with `tidal:` but not forming an uri but more like `tidal:artist:123` t.i. just a namespaced string.  (That's how I understand it, but I've never seen an `extid` in the wild tbf).

Reply:
Oh wow... that actually means you've found a bug: that regex should be able to parse those IDs. There's no use for it otherwise. I'll have to double check. But you're right, the external IDs we're using are `tidal:album:213` style.